operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java

```diff
@@ -16,82 +16,162 @@
+        baseDeployment.getSpec().setReplicas(keycloakCR.getSpec().getInstances());
+        baseDeployment.getSpec().getTemplate().getMetadata().setLabels(Constants.DEFAULT_LABELS);
+
+        Container container = baseDeployment.getSpec().getTemplate().getSpec().getContainers().get(0);
+        container.setImage(Optional.ofNullable(keycloakCR.getSpec().getImage()).orElse(config.operand().image()));
+
+        var distConfig = new HashMap<>(Constants.DEFAULT_DIST_CONFIG);
+        if (keycloakCR.getSpec().getDistConfig() != null) {
+            distConfig.putAll(keycloakCR.getSpec().getDistConfig());
+        }
+
+        container.setEnv(distConfig.entrySet().stream()
+                .map(e -> new EnvVarBuilder().withName(e.getKey()).withValue(e.getValue()).build())
+                .collect(Collectors.toList()));
+
+//        Set<String> configSecretsNames = new HashSet<>();
+//        List<EnvVar> configEnvVars = distConfig.entrySet().stream()
+//                .map(e -> {
+//                    EnvVarBuilder builder = new EnvVarBuilder().withName(e.getKey());
+//                    Matcher matcher = CONFIG_SECRET_PATTERN.matcher(e.getValue());
+//                    // check if given config var is actually a secret reference
+//                    if (matcher.matches()) {
+//                        builder.withValueFrom(
+//                                new EnvVarSourceBuilder()
+//                                        .withNewSecretKeyRef(matcher.group(2), matcher.group(1), false)
+//                                        .build());
+//                        configSecretsNames.add(matcher.group(1)); // for watching it later
+//                    } else {
+//                        builder.withValue(e.getValue());
+//                    }
+//                    builder.withValue(e.getValue());
+//                    return builder.build();
+//                })
+//                .collect(Collectors.toList());
+//        container.setEnv(configEnvVars);
+//        this.configSecretsNames = Collections.unmodifiableSet(configSecretsNames);
+//        Log.infof("Found config secrets names: %s", configSecretsNames);
+
+        return baseDeployment;
     }
 
-    public Deployment newKeycloakDeployment(Keycloak keycloak) {
-        if (baseDeployment == null) {
-            URL url = this.getClass().getResource("/base-deployment.yaml");
-            baseDeployment = client.apps().deployments().load(url).get();
+    public void updateStatus(KeycloakStatusBuilder status) {
+        if (existingDeployment == null) {
+            status.addNotReadyMessage("No existing Deployment found, waiting for creating a new one");
+            return;
         }
 
-        var deployment = baseDeployment;
+        var replicaFailure = existingDeployment.getStatus().getConditions().stream()
+                .filter(d -> d.getType().equals("ReplicaFailure")).findFirst();
```

Comment:
Just out of curiosity, any reason why are you re-mapping just this specific failure condition?

Reply:
Unfortunately, conditions are missing official documentation but to my understanding it is a good practice that K8s API server creates this condition when there are errors. Other conditions just tell us if we have all pods ready or if they're scaling. This one is about errors specifically.

Reply:
Thanks for explaining!